database/db.go

```diff
@@ -208,12 +210,18 @@ func NewDbFromConfig(c *Config, logger *logging.Logger, connectorCallbacks Retry
 			return nil, errors.Wrap(err, "can't open pgsql database")
 		}
 
-		addr = utils.JoinHostPort(c.Host, port)
 		db = sqlx.NewDb(sql.OpenDB(NewConnector(connector, logger, connectorCallbacks)), PostgreSQL)
+		addr = utils.JoinHostPort(c.Host, port)
```

Comment:
`c.Host` may also contain the path to a Unix domain socket for PostgreSQL. Have you tested that? In any case, adding a test for that would be nice.

Bonus points for correctly handling PostgreSQL's weirdness regarding Unix domain sockets (you configure `/run/postgresql` for example, but the actual socket is `/run/postgresql/.s.PGSQL.5432` and depends on the configured port number, so the port number is actually relevant in this case).

Reply:
Indeed, Unix domain support was missing for PostgreSQL. I have implemented the same quirky logic as for MySQL and have added a test.

> Bonus points for correctly handling PostgreSQL's weirdness

Sorry, but I don't want to implement anything more weird. The current state reflects the configuration as read from the configuration file and should be enough to show users where Icinga DB wants to connect to.

Reply:
> The current state reflects the configuration as read from the configuration file and should be enough to show users where Icinga DB wants to connect to.

Not entirely. Take the following config as an example. On might hope that the `port:` line is irrelevant, but in case of PostgreSQL, it isn't:

```yaml
database:
  type: pgsql
  host: /run/postgresql
  port: 3306
```

With just the address from this PR logged, that could give you some nasty debugging. However, the error message from `lib/pq` might actually save you as this includes the full socket path (output from the current Icinga DB `main` branch, that error should stay the same though):

```
WARN	database	Can't connect to database. Retrying	{"error": "dial unix /run/postgresql/.s.PGSQL.3306: connect: no such file or directory"}
```

> Sorry, but I don't want to implement anything more weird.

I mean it could be as simple as `(/run/postgresql):5432`. That wouldn't really add any complexity to the implementation.

Reply:
Fair enough, I have taken a closer look at the [documentaiton](https://www.postgresql.org/docs/17/runtime-config-connection.html#GUC-UNIX-SOCKET-DIRECTORIES) and decided to mimic their logic.